[Lee Wei Ming] iP

leeweiming3 opened the pull request:
# Pls acccept my pull request and rename it ran
- **Bullet**
- _List_
1. ~~Numbered~~
2. List
```css
.code :block {
syntax: "highlight";
}
```
- [x] Make a pull request
- [ ] :ribbon::eye_speech_bubble::ribbon:
- [x] :fox_face:
- [ ] :black_cat:
> Quote block



[Code](https://github.com/leeweiming3/ip/tree/master)
`print("TEST")`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

leeweiming3 committed:
Rename Duke to Ran

leeweiming3 committed:
add echo and exit functionality

leeweiming3 committed:
Added text recall function

leeweiming3 committed:
add to package, add class Task

leeweiming3 committed:
add task marking functionality and helper function parsenumber

leeweiming3 committed:
refactor task into another file and make chatbot an object

leeweiming3 committed:
fix formatting

leeweiming3 committed:
add deadline class and add parsing delimiters functionality

leeweiming3 committed:
add todos (fix regression)

leeweiming3 committed:
add event

leeweiming3 committed:
automated testing

leeweiming3 committed:
error handling was finished before

leeweiming3 committed:
refactor common methods

leeweiming3 committed:
add deletion

leeweiming3 committed:
fix plural

leeweiming3 committed:
Obey convention

leeweiming3 committed:
Add task save to file functionality

leeweiming3 committed:
Add tasklist object (automate saving)

leeweiming3 committed:
Merge branch 'branch-Level-7'

leeweiming3 committed:
Add date functionality

leeweiming3 committed:
Merge branch 'branch-Level-8'

leeweiming3 committed:
Unhardcode save location

leeweiming3 committed:
Use exceptions for errors and encapsulate UI

leeweiming3 committed:
Extract parser

leeweiming3 committed:
Error handle invalid dates

leeweiming3 committed:
Merge branch 'add-gradle-support' of https://github.com/leeweiming3/ip

leeweiming3 committed:
change class run by Gradle to Ran

leeweiming3 committed:
add tests for TaskList

leeweiming3 committed:
add javadocs for relevant classes

leeweiming3 committed:
obey convention

leeweiming3 committed:
rename TaskIO to TaskIo

leeweiming3 committed:
Add finding task function

leeweiming3 committed:
Merge branch 'branch-A-CodingStandard'

leeweiming3 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/ChatbotRan/RanUi.java

leeweiming3 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/ChatbotRan/Parser.java
#	src/main/java/ChatbotRan/Ran.java
#	src/main/java/ChatbotRan/RanUi.java

xiaorui-ui commented on src/main/java/ChatbotRan/Event.java:
```diff
@@ -0,0 +1,70 @@
+ * Events are Tasks that have a start and end date (LocalDate).
+ */
+public class Event extends Task {
+    private LocalDate from;
```
Perhaps you could name to 'start' and 'end'? 'from' and 'to' can cause confusion, just my personal opinion.

xiaorui-ui commented on src/main/java/ChatbotRan/RanUi.java:
```diff
@@ -0,0 +1,146 @@
+        System.out.println(task);
+    }
+
+    void printNumber(int size) {
```
Does size mean number of tasks in the list? Didn't seem apparent to me until I read the method implementation. 
```suggestion
    void printNumber(int listSize) {
```

xiaorui-ui commented on src/main/java/ChatbotRan/Parser.java:
```diff
@@ -0,0 +1,109 @@
+        }
+    }
+
+    private Task handleTaskNo(String line, int space, TaskList taskList) throws TaskException {
```
Does space mean position of space? 
```suggestion
    private Task handleTaskNo(String line, int posOfSpace, TaskList taskList) throws TaskException {
```

xiaorui-ui commented on src/main/java/ChatbotRan/RanUi.java:
```diff
@@ -0,0 +1,146 @@
+     *
+     * @param completed whether the task was complete
+     */
+    public void unmark(boolean completed) {
```
A boolean variable should sound like a boolean. 
```suggestion
    public void unmark(boolean isCompleted) {
```

xiaorui-ui commented on src/main/java/ChatbotRan/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+/**
+ * Stores and automatically saves/reads Tasks, using the given TaskIO.
+ */
+public class TaskList {
```
Good naming of methods!

xiaorui-ui commented on src/main/java/ChatbotRan/Util.java:
```diff
@@ -0,0 +1,38 @@
+    private Util() {
+    }
+
+    static Integer parseNumber(String line, int spacePos) {
```
Could you be consistent with the spacePos naming? It's quite easy to get confused.

xiaorui-ui commented on src/main/java/ChatbotRan/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @param t task
+     */
+    public void add(Task t) {
```
Perhaps use a more detailed name (e.g. = task) ?

xiaorui-ui reviewed (commented):
LGTM with regards to naming! Just a few comestic changes required.

jerremyng commented on src/main/java/ChatbotRan/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+        this.deadline = deadline;
+    }
+
+    public Deadline(String contents, String deadline) {
+        this(contents, LocalDate.parse(deadline));
```
I like this pattern of calling overloaded constructor in itself, great idea!

jerremyng commented on src/main/java/ChatbotRan/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    public LocalDate getDeadline() {
+        return deadline;
```
Maybe using this. might be clearer.
```suggestion
        return this.deadline;
```

jerremyng commented on src/main/java/ChatbotRan/Ran.java:
```diff
@@ -0,0 +1,56 @@
+ * Runs the entire chatbot. Reads from standard input.
+ */
+public class Ran {
+    Parser parser;
+    TaskList taskList;
+    TaskIo taskIo;
+    RanUi ui;
```
Would using access modifiers be appropriate here?

jerremyng commented on src/main/java/ChatbotRan/RanUi.java:
```diff
@@ -0,0 +1,146 @@
+     * Prints a generic error message.
+     */
+    public void unknown() {
+
```
Linebreak here might be unnecessary.

jerremyng commented on src/main/java/ChatbotRan/RanUi.java:
```diff
@@ -0,0 +1,146 @@
+        if (tasks.isEmpty()) {
+            System.out.println("No tasks contain that string.");
+        } else {
+            System.out.println("Found "+tasks.size()+" match"+(tasks.size()==1?"":"es"));
```
Perhaps use some spaces between operators to improve readability.
```suggestion
            System.out.println("Found " + tasks.size() + " match" + (tasks.size() == 1 ? "" : "es"));
```

jerremyng commented on src/main/java/ChatbotRan/Task.java:
```diff
@@ -0,0 +1,37 @@
+
+    public Task(String contents) {
+        this.contents = contents;
+        this.completed = false;
```
While completed might be clear that it is a boolean value, convention is to name such variables as isCompleted or hasCompleted
```suggestion
        this.isCompleted = false;
```

jerremyng reviewed (commented):
LGTM, great work and I see many clean coding patterns to learn from.

leeweiming3 committed:
add hello world javafx

leeweiming3 committed:
Add the rest of functionality in application

leeweiming3 committed:
Add error dialog

leeweiming3 committed:
Add javadocs for new javafx methods

leeweiming3 committed:
Update src/main/java/ChatbotRan/Parser.java

change space phrasing

Co-authored-by: xiaorui-ui <87259515+xiaorui-ui@users.noreply.github.com>

leeweiming3 committed:
Update src/main/java/ChatbotRan/Deadline.java

use this

Co-authored-by: jerremyng <100476104+jerremyng@users.noreply.github.com>

leeweiming3 committed:
Update src/main/java/ChatbotRan/RanUi.java

improve spacing

Co-authored-by: jerremyng <100476104+jerremyng@users.noreply.github.com>

leeweiming3 committed:
Fix parser and launch class

leeweiming3 committed:
Merge level 10

leeweiming3 committed:
Add some assertions

leeweiming3 committed:
Merge branch 'A-Assertions'

leeweiming3 committed:
Improve code quality

Proper spacing, consistent ordering of imports and final variables help future programmers understand your code.
Let's - Implement proper spacing conventions for improved readability, enforce consistent ordering of imports for uniformity and organize final variables

leeweiming3 committed:
Merge pull request #2 from leeweiming3/heads/A-CodeQuality

Improve code quality

leeweiming3 committed:
Add a title

leeweiming3 committed:
Add C-MassOps: mass mark, unmark and delete by specifying multiple tasks

leeweiming3 committed:
fix off-by-one error

leeweiming3 committed:
add ui mockup

leeweiming3 committed:
add user guide

leeweiming3 committed:
rename ui.png
